web/src/components/molecules/Asset/Asset/AssetBody/unzipFileList.tsx

```diff
@@ -1,76 +1,42 @@
 import { Key } from "rc-table/lib/interface";
-import { CSSProperties, useState } from "react";
+import { CSSProperties, useCallback, useEffect, useState } from "react";
 
 import Icon from "@reearth-cms/components/atoms/Icon";
 import Tree, { DataNode, TreeProps } from "@reearth-cms/components/atoms/Tree";
+import { AssetFile } from "@reearth-cms/components/molecules/Asset/asset.type";
 
 type Props = {
+  file: AssetFile;
   style?: CSSProperties;
 };
 
-// TODO: this data is hard-codded, actual data should be formatted in a similar way.
-const treeData: DataNode[] = [
-  {
-    title: "Folder",
-    key: "0-0",
-    children: [
-      {
-        title: "Folder",
-        key: "0-0-0",
-        children: [
-          {
-            title: "File",
-            key: "0-0-0-0",
-          },
-          {
-            title: "File",
-            key: "0-0-0-1",
-          },
-          {
-            title: "File",
-            key: "0-0-0-2",
-          },
-        ],
-      },
-      {
-        title: "Folder",
-        key: "0-0-1",
-        children: [
-          {
-            title: "File",
-            key: "0-0-1-0",
-          },
-        ],
-      },
-      {
-        title: "Folder",
-        key: "0-0-2",
-        children: [
-          {
-            title: "File",
-            key: "0-0-2-0",
-          },
-          {
-            title: "File",
-            key: "0-0-2-1",
-          },
-        ],
-      },
-    ],
-  },
-  {
-    title: "File",
-    key: "0-1",
-  },
-  {
-    title: "File",
-    key: "0-2",
-  },
-];
-
-const UnzipFileList: React.FC<Props> = ({ style }) => {
-  const [expandedKeys, setExpandedKeys] = useState<DataNode["key"][]>(["0-0"]);
+const UnzipFileList: React.FC<Props> = ({ file, style }) => {
+  const [expandedKeys, setExpandedKeys] = useState<DataNode["key"][]>(["0"]);
   const [selectedKeys, setSelectedKeys] = useState<DataNode["key"][]>([]);
+  const [treeData, setTreeData] = useState<DataNode[]>([]);
+
+  const getTreeData = useCallback((assetFile: AssetFile, key = ""): DataNode[] => {
+    return (
+      assetFile?.children?.map((file: AssetFile, index: number) => {
+        let children: DataNode[] = [];
+        if (file.children && file.children.length > 0) {
+          children = getTreeData(
+            file,
+            key === "" ? index.toString() : key + "-" + index.toString(),
+          );
+        }
+        return {
+          title: file.name,
+          key: key === "" ? index.toString() : key + "-" + index.toString(),
+          children: children,
+        };
+      }) || []
+    );
+  }, []);
```

Comment:
Should be able to write this a bit cleaner like (test it out before committing it since there might be a typo somewhere/formatting issues):

```suggestion
  const getTreeData = useCallback((assetFile: AssetFile, key = ""): DataNode[] => assetFile?.children?.map((file: AssetFile, index: number) => {
        let children: DataNode[] = [];
        if (file.children && file.children.length > 0) {
          children = getTreeData(
            file,
            key === "" ? index.toString() : key + "-" + index.toString(),
          );
        }
        return {
          title: file.name,
          key: key === "" ? index.toString() : key + "-" + index.toString(),
          children: children,
        };
      }) || [], []);
```

Reply:
I couldn't find the type, however I did the suggested change.

Reply:
How about this?
```ts
  const getTreeData = useCallback(
    (assetFile: AssetFile, key: string): DataNode[] =>
      assetFile?.children?.map((file: AssetFile, index: number) => {
        return {
          title: file.name,
          key: key === "" ? index.toString() : key + "-" + index.toString(),
          children:
            file.children && file.children.length > 0
              ? getTreeData(file, key === "" ? index.toString() : key + "-" + index.toString())
              : [],
        };
      }) || [],
    [],
  );
```